packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -5203,6 +5190,26 @@ class BaseModelSqlv2 {
     return await this.execAndParse(qb);
   }
 
+  public async execAndGetRows(query: string, trx?: Knex | CustomKnex) {
+    trx = trx || this.dbDriver;
+
+    query = this.sanitizeQuery(query);
+
+    if (this.isPg || this.isSnowflake) {
+      return (await trx.raw(query))?.rows;
+    } else if (/^(\(|)select/i.test(query) && !this.isMssql) {
+      return await trx.from(trx.raw(query).wrap('(', ') __nc_alias'));
+    } else if (/^(\(|)insert/i.test(query) && this.isMySQL) {
```

Comment:
one improvement we can do here to avoid unnecessary regex match

```
    } else if ( !this.isMssql && /^(\(|)select/i.test(query)) {
      return await trx.from(trx.raw(query).wrap('(', ') __nc_alias'));
    } else if (this.isMySQL && /^(\(|)insert/i.test(query)) {
```